src/paste3/paste.py

```diff
@@ -104,9 +104,6 @@ def pairwise_align(
     coordinatesB = sliceB.obsm["spatial"].copy()
     coordinatesB = nx.from_numpy(coordinatesB)
 
-    if isinstance(nx, ot.backend.TorchBackend):
```

Comment:
I suspect this code was put in ad-hoc because the math operations weren't being called correctly because of dtypes being mixed when using `TorchBackend` (numpy is more forgiving of mixing dtypes than torch).

This code was the first indication why tests were failing on macs (reducing precision this way will cause platform-specific differences to magnify). Thus these kinds of `.float()` conversions (at 3 places) have been jettisoned.